### PR TITLE
[v4] Removed test errors / warning

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### WHY are these changes introduced?

We're starting to get a bunch of errors, and non-deprecation warning in our tests and they're being very noisy 🙅‍♂ 

We currently have two errors left. One is from enzyme that will hopefully be fixed eventually 🤞 and the other is Tabs Item which is placing a ref on a functional component. We can either use forwardRef on UnstyledLink or find an alternative using a ref on a component. Which never would have allowed us to call `.focus` afaik. We'll have a reference to the component so [this](https://github.com/Shopify/polaris-react/blob/77ea29e563ce826870acd090bfed1b24973b8c7a/src/components/Tabs/components/Item/Item.tsx#L23) will never be true. I'll create a follow for us to follow up on this.

### WHAT is this pull request doing?

* adjusting tests by adding mocks or missing handlers

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,7 +105,7 @@
     "@storybook/addon-viewport": "5.1.0-rc.0",
     "@storybook/react": "5.1.0-rc.0",
     "@storybook/theming": "5.1.0-rc.0",
-    "@types/enzyme": "^3.9.1",
+    "@types/enzyme": "^3.9.3",
```
Comment (AndrewMusgrave) [inline package.json]: In the end, this upgrade didn't fix the enzyme bug but I'm keeping it in to keep us up to date

File: src/components/Frame/components/Toast/tests/Toast.test.tsx [hunk 2]
```diff
@@ -94,14 +94,16 @@ describe('<Toast />', () => {
     });
 
     it('warns that a duration of 10000ms is recommended with action if duration is lower than 10000', () => {
-      const warnSpy = jest.spyOn(console, 'log');
+      const logSpy = jest.spyOn(console, 'log');
```
Comment (AndrewMusgrave) [inline src/components/Frame/components/Toast/tests/Toast.test.tsx]: log -> warn 😄

File: src/components/Tabs/components/Item/Item.tsx [hunk 3]
```diff
@@ -55,7 +55,9 @@ export default class Item extends React.PureComponent<Props, never> {
     };
 
     const markup = url ? (
-      React.createElement(UnstyledLink, {url, ...sharedProps}, children)
+      <UnstyledLink {...sharedProps} url={url}>
```
Comment (AndrewMusgrave) [inline src/components/Tabs/components/Item/Item.tsx]: This didn't fix the error but we should have been doing this, to begin with. IIFC we had a type issue in the past forcing us to use `createElement` that's no longer present.

**Comment by BPScott** [comment]
What here is v4 specific? I can't see a reason why this shouldn't be put on master instead?

**Comment by AndrewMusgrave** [comment]
> What here is v4 specific? I can't see a reason why this shouldn't be put on master instead?

@BPScott About half of the errors started to occur with the react version bump

**Pull request merged**